update the documentation for the _collection init_ command (#78404)
ansible/ansible: docs/docsite/rst/dev_guide/developing_collections_creating.rst

Creating a collection skeleton
==============================

> To start a new collection run in your collections directory:

Reviewer: ```suggestion
To start a new collection, run the following command in your collections directory:
```

.. code-block:: bash

>     ansible_collections#> ansible-galaxy collection init my_namespace.my_collection

Reviewer: Not sure if this is really needed.

Reviewer: @s-hertel thoughts?

Reviewer 2: I personally thought this change made it more clear that the collection my_namespace.my_collection is getting initialized in a directory that can contain multiple collections.

>     ansible_collections#> ansible-galaxy collection init my_namespace.my_collection

Reviewer 2: I personally thought this change made it more clear that the collection my_namespace.my_collection is getting initialized in a directory that can contain multiple collections.

Reviewer: okay. resolving.

.. note::

	Both the namespace and collection names use the same strict set of requirements. See `Galaxy namespaces <https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespaces>`_ on the Galaxy docsite for those requirements.

> It will create the structure ``[my_namespace]/[my_collection]/[collection skeleton]``

Reviewer: ```suggestion
It will create the structure ``[my_namespace]/[my_collection]/[collection skeleton]``.
```

> The git project must be initialized in the collection directory.

Reviewer: There is no obligation to use Git or to version-control the source code at all. So using "must" seems misleading at best. Not sure if Git even deserves to be mentioned in such a context. Maybe some `.. hint::` admonition could be useful, though.

Author: Agree, changed.

> .. hint:: The git project should be initialized in the collection directory.

Reviewer: Have you checked if it's rendered correctly, I'm pretty sure it needs empty lines around it like `.. note::` above:
```suggestion

.. hint::

   If Git is used for version control, the corresponding repository should be initialized in the collection directory.

```

Author: I have not checked the render, but the 7 places where the `.. hint::` is used ini the project is without empty lines. 
```sh
grep -r ".. hint::" .
./docs/docsite/rst/scenario_guides/guide_meraki.rst:.. hint:: Use the ``org_id`` and ``net_id`` parameters when possible. ``org_name`` and ``net_name`` require additional behind-the-scenes API calls to learn the ID values. ``org_id`` and ``net_id`` will perform faster. 
./docs/docsite/rst/scenario_guides/guide_aci.rst:.. hint:: For clarity we have added ``delegate_to: localhost`` to all the examples in the module documentation. This helps to ensure first-time users can easily copy&paste parts and make them work with a minimum of effort.
./docs/docsite/rst/scenario_guides/guide_aci.rst:.. hint:: Selective proxy support using the ``no_proxy`` environment variable is also supported.
./docs/docsite/rst/scenario_guides/guide_aci.rst:.. hint:: If you use a certificate name in ACI that matches the private key's basename, you can leave out the ``certificate_name`` parameter like the example above.
./docs/docsite/rst/scenario_guides/guide_aci.rst:.. hint:: The XML format is more practical when there is a need to template the REST payload (inline), but the YAML format is more convenient for maintaining your infrastructure-as-code and feels more naturally integrated with Ansible playbooks. The dedicated modules offer a more simple, abstracted, but also a more limited experience. Use what feels best for your use-case.
./docs/docsite/rst/dev_guide/developing_collections_creating.rst:.. hint:: The git project should be initialized in the collection directory.
./docs/docsite/rst/dev_guide/developing_modules_general_aci.rst:.. hint:: Do not provide default values for configuration arguments. Default values could cause unintended changes to the object.
```

Author: Anyway, I have updated the hint with your suggestion (but no empty lines added)
